pkg/policy/api/rule_validation.go

```diff
@@ -381,6 +381,41 @@ func (pr *PortRule) sanitize(ingress bool) error {
 		}
 	}
 
+	listener := pr.Listener
+	if listener != nil {
+		if listener.Name == "" {
+			return fmt.Errorf("Listener must have name")
+		}
+		if listener.EnvoyConfig == nil {
+			return fmt.Errorf("Listener must have envoyConfig (%s)",
+				listener.Name)
+		}
+		if listener.EnvoyConfig.Name == "" {
+			return fmt.Errorf("Listener envoyConfig must have name (%s)", listener.Name)
+		}
+
+		switch listener.EnvoyConfig.Kind {
+		case "CiliumEnvoyConfig", "CiliumClusterwideEnvoyConfig", "":
+		default:
+			return fmt.Errorf("Listener envoyConfig.kind must be CiliumEnvoyConfig, CiliumClusterwideEnvoyConfig, or left empty (%s)", listener.Name)
```

Comment:
This error msg suggests that empty is not a valid `kind`, but the case above handles it. Also, the godoc on the `Kind` type above also suggests that it's optional (which means the field can be empty). So which is it? :slightly_smiling_face:

Reply:
Empty is allowed, with sane defaulting as I think I wrote somewhere.